### ISSUE TITLE
Define ForallF in terms of Forall?

### ISSUE THREAD

**Issue by treeowl** [ekmett/constraints#23]
Now that `Forall` is poly-kinded, I think we can define `ForallF` to use it:

```haskell
class p (f a) => ComposeC (p :: k2 -> Constraint) (f :: k1 -> k2) (a :: k1)
instance p (f a) => ComposeC p f a

class Forall (ComposeC p f) => ForallF (p :: k2 -> Constraint) (f :: k1 -> k2)
instance Forall (ComposeC p f) => ForallF p f

instF :: forall p f a . ForallF p f :- p (f a)
instF = Sub $
  case inst :: Forall (ComposeC p f) :- ComposeC p f a of
   Sub Dict -> Dict
```

Unfortunately, I haven't been able to find a way to write functions converting between this `ForallF` and the standard `ForallF` without mentioning the names of skolems. The essential difficulty seems to be moving `forall`s around. I conjecture, however, that they are fundamentally equivalent, and that anything in the wild that satisfies one will also satisfy the other.

**Comment by ekmett** [comment]
Not a bad plan. This is more or less how I implement `Lim` in ekmett/hask

**Comment by treeowl** [comment]
Implemented by #24.
